.tekton/build_and_trigger.yaml

```diff
@@ -0,0 +1,109 @@
+      - name: repo_url
+    workspaces:
+      - name: workspace
+    tasks:
+      - name: fetch-repo
+        workspaces:
+          - name: output
+            workspace: workspace
+        taskRef:
+          name: git-clone
+        params:
+          - name: url
+            value: "$(params.repo_url)"
+          - name: revision
+            value: "$(params.revision)"
+      - name: build-image
+        workspaces:
+          - name: source
+            workspace: workspace
+        params:
+          - name: IMAGE
+            value: quay.io/redhat-appstudio/clair-in-ci:{{revision}}
+          - name: BUILDER_IMAGE
+            value: $(params.builder-image)
+          - name: DOCKERFILE
+            value: $(workspaces.source.path)/$(params.dockerfile)
+        runAfter:
+          - fetch-repo
+        taskRef:
+          name: buildah
+          kind: ClusterTask
+      - name: trigger-tests-and-push
+        params:
+          - name: revision
+            value: "{{ revision }}"
+          - name: target_branch
+            value: "{{ target_branch }}"
+          - name: source_branch
+            value: "{{ source_branch }}"
+        runAfter:
+          - build-image
+        taskSpec:
+          params:
+            - name: revision
+              type: string
+            - name: target_branch
+              type: string
+            - name: source_branch
+              type: string
+          steps:
+            - name: tigger-workflow
+              image: registry.access.redhat.com/ubi9/python-39
+              env:
+                - name: GITHUB_TOKEN
+                  valueFrom:
+                    secretKeyRef:
+                      name: integration-github-token
+                      key: integration-github-token
+              script: |
+                #!/usr/bin/env bash
+                if [ "$(params.target_branch)" == "daily" ]; then
+                  curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_TOKEN" --request POST --data '{"event_type":"trigger_workflow", "client_payload": {"event": "daily", "image_tag": "$(params.revision)"}}' https://api.github.com/repos/redhat-appstudio/clair-in-ci-db/dispatches
+                elif [ "$(params.target_branch)" == "main" ] && [ "$(params.source_branch)" == "main" ]; then
+                  curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_TOKEN" --request POST --data '{"event_type":"trigger_workflow", "client_payload": {"event": "push", "image_tag": "$(params.revision)"}}' https://api.github.com/repos/redhat-appstudio/clair-in-ci-db/dispatches
+                else
+                  curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_TOKEN" --request POST --data '{"event_type":"trigger_workflow", "client_payload": {"event": "pull_request", "image_tag": "$(params.revision)"}}' https://api.github.com/repos/redhat-appstudio/clair-in-ci-db/dispatches
+                fi
```

Comment:
```suggestion
                event="pull_request"
                if [ "$(params.target_branch)" == "daily" ]; then
                  event="daily"
                elif [ "$(params.target_branch)" == "main" ] && [ "$(params.source_branch)" == "main" ]; then
                  event="push"
                fi
                curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_TOKEN" --request POST --data '{"event_type":"trigger_workflow", "client_payload": {"event": "${event}", "image_tag": "$(params.revision)"}}' https://api.github.com/repos/redhat-appstudio/clair-in-ci-db/dispatches
```

Reply:
tekton-ci doesn't have {{ event }} parameter accessible outside of annotations.

Reply:
I defined it, it's not taken from tekton.
I just rewrote your code to have minimal critical section